llvm/lib/Analysis/LoopAccessAnalysis.cpp

```diff
@@ -2061,14 +2061,12 @@ MemoryDepChecker::getDependenceDistanceStrideAndSize(
     return MemoryDepChecker::Dependence::Unknown;
   }
 
-  TypeSize AStoreSz = DL.getTypeStoreSize(ATy);
-  TypeSize BStoreSz = DL.getTypeStoreSize(BTy);
-
-  // If store sizes are not the same, set TypeByteSize to zero, so we can check
-  // it in the caller isDependent.
   uint64_t ASz = DL.getTypeAllocSize(ATy);
   uint64_t BSz = DL.getTypeAllocSize(BTy);
-  uint64_t TypeByteSize = (AStoreSz == BStoreSz) ? BSz : 0;
+
+  // The TypeByteSize is used to scale Distance and VF. In these contexts, the
+  // only size that matters is the size of the Sink.
+  uint64_t TypeByteSize = BSz;
```

Comment:
It's not clear to me why picking the size of the sink is always correct. I've not checked all uses, but intuitively the larger size may serve as an upper bound for some (all?) uses

Reply:
So, TypeByteSize is used in couldPreventStoreLoadForward, and in the MinDistanceNeeded computation in the same way: it is used to scale distance and VF; isn't the size of the sink sufficient? It is also used in areStridedAccessesIndependent, where Distance % TypeByteSize should be zero: isn't the size of the sink sufficient here as well? Not sure why the size of the source should be considered in these cases.